client/src/utils/getCompressParams.ts

```diff
@@ -15,7 +15,7 @@ export const getCompressParams = (
     },
   ];
 
-  let archivePath = `${files[0].path.slice(0, -files[0].name.length)}`;
+  let archivePath = `${files[0].path.slice(0, -files[0].name.length)}` || '.'; // set archive path to root if no enclosing folder
```

Comment:
👍 

suggestion:  could the calculation of archivePath (here and in `getExtractParams`) be extracted to a standalone method (`getParentPathForArchiving` or `getParentPath`) that lives in jobsUtil.js.

 It would reduce duplication and allow for some extra unit tests that cover the main scenarios


```
describe('getParentPath', () => {
  it('returns directory path when file is in a folder', () => {
    const file = { path: 'some/folder/test.txt', name: 'test.txt' };
    expect(getParentPath(file)).toEqual('some/folder/');
  });

  it('returns "." when file is at root level', () => {
    const file = { path: 'test.txt', name: 'test.txt' };
    expect(getParentPath(file)).toEqual('.');
  });
}

Reply:
Yeah, I can definitely do this! And I can move the tests to `jobsUtil.test.js` to reduce duplicate tests.